### PR TITLE
Alert Success/Delete UI

### PR DESCRIPTION
**Pull request by veverlymae**
# 📲 What
- Alert Success/Delete UI

# 🤔 Why
- Create Reusable AlertButton, AlertMessage and AlertImageView
- Create Alert View for Success/Delete

# 🛠 How
- N/A

# ✅ Acceptance criteria
- N/A

# ⏰ TODO
-N/A

# 🔗 References
https://xd.adobe.com/view/88019197-b52c-4c03-a717-e10207152dd7-dd52/screen/4e1b839f-e7e8-46c1-9d45-5719013ffadb/
https://xd.adobe.com/view/88019197-b52c-4c03-a717-e10207152dd7-dd52/screen/7f516a7b-f26e-4727-9f2e-05827057fbe8/
Screen Shot 2021-05-05 at 6 39 29 PM

### CODE REVIEW

File: Kwartracker/Scenes/Reusable/ButtonView.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    
+    var body: some View {
+        Button(action: actionHandler, label: {
+            ZStack {
+                BWNeumorphicRectangle(rectRadius: rectRadius,
+                                      color: Color(Asset.Colors.blueberry.color),
+                                      shadowRadius: shadowRadius,
+                                      shadowOffset: shadowOffset)
+                
+                HStack {
```
Comment (jaimejazarenoiii) [inline Kwartracker/Scenes/Reusable/ButtonView.swift:27]: I think HStack here is unnecessary

File: Kwartracker/Scenes/Reusable/ImageView.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    
+    let imageType: ImageType
+    
+    var body: some View {
+        VStack{
+            Image(uiImage: imageType.image)
+        }
+    }
+    
+    enum ImageType {
```
Comment (jaimejazarenoiii) [inline Kwartracker/Scenes/Reusable/ImageView.swift:20]: Actually all of these enums with success and delete can be refactored into just one enum, maybe DialogState? then you'll have two options, either make the message included in the cases or create another params per view with an additional param `message`

File: Kwartracker/Scenes/Alert/AlertView.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+//
+
+import SwiftUI
+
+struct AlertView: View {
+    var body: some View {
+        ZStack {
+            Color.gray.opacity(0.4).edgesIgnoringSafeArea(.all)
+            MainAlertView()
+                .frame(width: 220, height: 240, alignment: .center)
```
Comment (jaimejazarenoiii) [inline Kwartracker/Scenes/Alert/AlertView.swift:15]: If the alert is uniform in size, move the magic numbers to a constant, and I think the height should be dynamic not fixed

File: Kwartracker/Scenes/Alert/ Content/MainAlertView.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+//  MainAlertView.swift
+//  Kwartracker
+//
+//  Created by Veverly Mae Ricaza Veverly on 4/27/21.
+//
+
+import SwiftUI
+
+struct MainAlertView: View {
+
```
Comment (jaimejazarenoiii) [inline Kwartracker/Scenes/Alert/ Content/MainAlertView.swift:11]: maybe move the params here

state: DialogStatus, message: String

File: Kwartracker/Scenes/Alert/ Content/MainAlertView.swift [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import SwiftUI
+
+struct MainAlertView: View {
+    
+    var body: some View {
+        VStack {
+            ImageView(imageType: .success)
+            MessageView(messageType: .success)
+            ButtonView(actionHandler: {}, actionLabel: .success)
```
Comment (jaimejazarenoiii) [inline Kwartracker/Scenes/Alert/ Content/MainAlertView.swift:16]: Can create a style for the view depending on the status

File: Kwartracker/Scenes/Reusable/MessageView.swift [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        }
+        
+        var message: String {
+            switch self {
+            case .success:
+                return "New category has been added."
+            case .deleted:
+                return "It has been successfully deleted."
+            case .confirmation:
+                return "Are you sure you want to delete this?"
```
[the remark this answers is not in the document]
Comment (jaimejazarenoiii) [reply Kwartracker/Scenes/Reusable/MessageView.swift:51]: I think this should be dynamic

**Review by jaimejazarenoiii — CHANGES_REQUESTED** [review]

**Review by ljylaya — APPROVED** [review]

**Pull request merged**